collect_app/src/main/java/org/odk/collect/android/formentry/FormEntryViewModel.java

```diff
@@ -35,14 +40,20 @@
     private FormIndex jumpBackIndex;
 
     @SuppressWarnings("WeakerAccess")
-    public FormEntryViewModel(Clock clock, Analytics analytics) {
+    public FormEntryViewModel(Clock clock, Analytics analytics, PreferencesProvider preferencesProvider, AudioRecorder audioRecorder) {
         this.clock = clock;
         this.analytics = analytics;
+        this.preferencesProvider = preferencesProvider;
+        this.audioRecorder = audioRecorder;
     }
 
     @Override
     public void formLoaded(@NotNull FormController formController) {
         this.formController = formController;
+
+        if (hasBackgroundRecording()) {
```

Comment:
Felt like this made more sense as a place to start the recording. I'm also trying to hide away the fact that this is triggered by prefs as much as possible.